### DISCUSSION TITLE
[Turbo] - The "use server" directive must be at the top of the file.

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [vercel/next.js#64863] [Help]
### Summary

Hey, i was trying out turbo on a repo for work, which runs fine normally, when moving over to turbo i get this error on dev

Build Error
Failed to compile

Next.js (14.3.0-canary.15) (turbo)
./src/components/SearchBox/functions.ts:1:1
Ecmascript file had an error
> 1 | "use server";
    | ^^^^^^^^^^^^
  2 |
  3 | import useQuery from "@/hooks/useQuery";
  4 | import { type Listing, type SearchPagination } from "@/utils/types";

The "use server" directive must be at the top of the file.

### Additional information

```js
"use server";

import useQuery from "@/hooks/useQuery";
import { type Listing, type SearchPagination } from "@/utils/types";

import { GET_LISTINGS } from "../../app/graphql";

export const getSearchSuggestions = async (query: string) => {
  const { data } = await useQuery<SearchPagination<"listings_search", Listing>>(
    GET_LISTINGS,
    {
      variables: {
        filter: {
          status: "publish",
        },

        search: {
          text: {
            query,
          },
        },
      },
    },
  );

  return { data };
};


This is the code that is erroring, this function is imported into a client search component.

Any reason why this is failing?

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Wed Feb 21 21:45:49 PST 2024; root:xnu-10063.101.15~2/RELEASE_ARM64_T6020
  Available memory (MB): 16384
  Available CPU cores: 12
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 3.6.1
  pnpm: 8.6.12
Relevant Packages:
  next: 14.3.0-canary.15 // Latest available version is detected (14.3.0-canary.15).
  eslint-config-next: 14.1.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.3
Next.js Config:
  output: N/A
```

**Comment by kyrregjerstad** [comment]
I'm having the same issue, did you find any workaround for this?

**Comment by polaris0211** [comment]
My code makes same issue

**Comment by Dxuian** [comment]
this issue is still persisting next@14.2.3 any workaround or permanent fixes ?  
ill be raising this discussion to issue as i havent found a fix

**Comment by Antonytm** [comment]
Same for me.

**Comment by harisiqbal12** [comment]
Same issue with my project as well.

**Comment by Youssef-Hassane** [comment]
+1 🤡🤡

**Comment by NikkingUA** [comment]
I think If useQuery have "use client"??

**Comment by slimbde** [comment]
Server components are not allowed to use any hooks. Get rid of **useQuery** and it would do.

**Reply by MiroslavPetrik** [reply]
This could be it.
I had one function `useTranslation` which was named like a hook, but used in server components and server action.
After renaming the "hook" to just `translate` it got rid of the error:

https://github.com/MiroslavPetrik/net3-starter/commit/d36c1f59540b433176fc9d59ba302880ef2ac3d3#r149933209

**Reply by mikemasam** [reply]
@MiroslavPetrik same, it seem the useHook naming causes this

**Reply by TJBlackman** [reply]
Fixed my issue. I wasn't even using a hook, but the function I was using was following the `useThing` name scheme. I changed it at the time of import and it works `import { useThing as getThing } from '../source'`

**Reply by JoaoVictorCabraldeMelo** [reply]
If someone are using useTranslations on server-side for me it solved replacing with

`import { getTranslations } from 'next-intl/server';`

**Comment by wmeints** [comment]
I had the same issue, I named a variable `useCase` which triggers this error. Not a very descriptive error. I would expect it to point me to my variable and yell at me that I should not have `use[....]` in my code anywhere.
